code/parachain/frame/cosmwasm/Cargo.toml

```diff
@@ -5,6 +5,9 @@ homepage = "https://composable.finance"
 name = "pallet-cosmwasm"
 version = "1.0.0"
 
+[lib]
+doctest = false
```

Comment:
because nobody maintains them anyway, the are broken and not run on CI. so until tests are fixed or run, let default runs to not fail.

Reply:
if somebody will have time to fix doc tests, one can do that.

Reply:
If they are not run on CI why disable them? If you want to run tests without dectests locally just run `cargo test --lib`.  Disabling tests creates an illusion that the test are working and there’s a risk someone will add a new doctest thinking it verifies their code.

Reply:
i run them locally. i will not going to fix them, write new or maintain. are you up fix these?

Reply:
tbh it does not creates illusion - it very stratforawdly says - writes if you want doc tests, but i do not care. 

i care about now failing e2e simulation test. really we need only one simulation test which must for XCVM/intetion deliver i guess.

what are you are reffering is part of this https://github.com/w3f/Grants-Program/pull/1430  - having doc tests.  I guess you may talk to Jafar about if we should handle that now or later.

Reply:
> If they are not run on CI why disable them? 

They run on CI btw - i enabled in this PR. Simulation tests fails - i retained it as going eventually fix it.

Reply:
> there’s a risk someone will add a new doctest thinking it verifies their code.

1. so he clear that CI does not cares and some people about these docs tests
2. he can add normal tests whcih btw easier to maintain rahter than /// test
3. no really sure we need so many small tests now (see https://github.com/w3f/Grants-Program/pull/1430  )

Reply:
> i run them locally

Yes, and you can continue doing that with `cargo test --lib`:

```console
…code/parachain/frame/cosmwasm$ cargo test --lib
    Finished test [unoptimized] target(s) in 0.55s
     Running unittests src/lib.rs (/srv/mpn/comp/composable/code/target/debug/deps/pallet_cosmwasm-66c0d37fb7819119)

running 36 tests
...
test benchmarking::benchmark_tests::test_benchmarks ... ok

test result: ok. 36 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 4.32s

…code/parachain/frame/cosmwasm$
```

> tbh it does not creates illusion - it very stratforawdly says - writes if you want doc tests, but i do not care.

It does.  You might not care but someone who writes a doctest cares and expects about it and expects that if they run `cargo test` it will execute that test.

> he can add normal tests whcih btw easier to maintain rahter than /// test

That depends.  Doctests live right next to the code that is being tested so it’s easier to edit both together if needed.  Furthermore, advantage of doctests is that they provide example usage and thus serve as documentation (something that the code base is *severely* lacking and we should rather go into direction of improving that situation).

> no really sure we need so many small tests now (see https://github.com/w3f/Grants-Program/pull/1430 )

I’m not sure how this PR is relevant here? Small tests are useful because they help catch bugs earlier and faster.  If a big integration tests fails you need to spend time pin-pointing what part of the code failed.  So I don’t know what you mean by not needing small tests to be honest.

Reply:
https://github.com/ComposableFi/composable/pull/3683

Reply:
> Yes, and you can continue doing that with cargo test --lib:

I would have to modify CI. I want CI to be cohesive along codebase.